uber/custom_tags.py

```diff
@@ -388,253 +274,138 @@ def extract_fields(what):
-            'charge': charge
-        }
+def normalize_newlines(text):
+    return re.sub(r'\r\n|\r|\n', '\n', text)
 
-        return render('preregistration/stripeForm.html', params)
 
+@JinjaEnv.jinja_export
+def csrf_token():
+    if not cherrypy.session.get('csrf_token'):
+        cherrypy.session['csrf_token'] = uuid4().hex
+    return safe_string('<input type="hidden" name="csrf_token" value="{}" />'.format(cherrypy.session["csrf_token"]))
 
-@register.tag('bold_if')
-def do_bold_if(parser, token):
-    [cond] = token.split_contents()[1:]
-    nodelist = parser.parse(('end_bold_if',))
-    parser.delete_first_token()
-    return BoldIfNode(cond, nodelist)
 
+@JinjaEnv.jinja_export
+def stripe_form(action, charge):
+    payment_id = uuid4().hex
+    cherrypy.session[payment_id] = charge.to_dict()
 
-class BoldIfNode(template.Node):
-    def __init__(self, cond, nodelist):
-        self.cond = Variable(cond)
-        self.nodelist = nodelist
+    email = ''
+    if charge.targets and charge.models[0].email:
+        email = charge.models[0].email[:255]
 
-    def render(self, context):
-        cond = self.cond.resolve(context)
-        output = self.nodelist.render(context)
-        if cond:
-            return '<b>' + output + '</b>'
+    if not charge.targets:
+        if c.AT_THE_CON:
+            regtext = 'On-Site Charge'
         else:
-            return output
-
-
-@tag
-class organization_and_event_name(template.Node):
-    def render(self, context):
-        if c.EVENT_NAME.lower() != c.ORGANIZATION_NAME.lower():
-            return c.EVENT_NAME + ' and ' + c.ORGANIZATION_NAME
-        else:
-            return c.EVENT_NAME
-
-
-@tag
-class organization_or_event_name(template.Node):
-    def render(self, context):
-        if c.EVENT_NAME.lower() != c.ORGANIZATION_NAME.lower():
-            return c.EVENT_NAME + ' or ' + c.ORGANIZATION_NAME
-        else:
-            return c.EVENT_NAME
-
+            regtext = 'Charge'
+    elif c.AT_THE_CON:
+        regtext = 'Registration'
+    else:
+        regtext = 'Preregistration'
 
-@tag
-class single_day_prices(template.Node):
-    def render(self, context):
-        prices = ''
-        for day, price in c.BADGE_PRICES['single_day'].items():
-            if day == datetime.strftime(c.ESCHATON, "%A"):
-                prices += 'and ${} for {}'.format(price, day)
-                break
-            else:
-                prices += '${} for {}, '.format(price, day)
-        # prices += 'and ${} for other days'.format(c.BADGE_PRICES['default_single_day'])
-        return prices
+    params = {
+        'action': action,
+        'regtext': regtext,
+        'email': email,
+        'payment_id': payment_id,
+        'charge': charge
+    }
 
+    # TODO: not 100% sure this is kosher with the way we're doing our
+    # singleton JinjaEnv() class.  might have to make this not a singleton
+    return render('preregistration/stripeForm.html', params).decode('utf-8')
```

Comment:
we should clear this TODO comment or decide to not do this.

this comment is basically saying that:
1) we're inside a Jinja render currently
2) we're calling render() again inside here which starts a new render in our global Jinja2 singleton.  after we get the html output, we return that as the output of this tag
3) we're not sure that this is OK.  it seems to work fine?  i think it's OK?  it seems like jinja would be built to be able to handle this case? just not 100% sure.  we did this kind of thing with django just fine